Unexpected file size error being thrown when reading large file

<details>
<summary>Stack trace</summary>
```
Reading video from: /workspace/template-helloworld/out/HelloWorld.gif
{"level":"info","msg":"Command executed completed successfully","component":"sandbox-do","sandboxId":"747f7eee07b30243447c9845f40ca8064c4b3e112e06f821ea107ef045a2cda9","traceId":"tr_3b30b96af2674e30","details":"ffprobe -v error -show_entries stream=profile,pix_fmt,codec_name -of default=noprint_wrappers=1 \"/workspace/template-helloworld/out/HelloWorld.gif\", Success: true","timestamp":"2025-11-26T12:28:26.133Z"}
Video codec info: codec_name=gif
profile=unknown
pix_fmt=bgra
Conversion disabled, uploading original video
{"level":"info","msg":"Command executed completed successfully","component":"sandbox-do","sandboxId":"747f7eee07b30243447c9845f40ca8064c4b3e112e06f821ea107ef045a2cda9","traceId":"tr_3b30b96af2674e30","details":"ls -lh \"/workspace/template-helloworld/out/HelloWorld.gif\", Success: true","timestamp":"2025-11-26T12:28:26.206Z"}
File info: -rw-r--r-- 1 root root 17M Nov 26 12:28 /workspace/template-helloworld/out/HelloWorld.gif
Attempting to read file with readFile (base64)...
✘ [ERROR] {"level":"error","msg":"Unexpected error in readFile","component":"sandbox-do","sandboxId":"747f7eee07b30243447c9845f40ca8064c4b3e112e06f821ea107ef045a2cda9","traceId":"tr_3b30b96af2674e30","httpStatus":500,"timestamp":"2025-11-26T12:28:27.073Z","error":{"message":"Failed to read binary file '/workspace/template-helloworld/out/HelloWorld.gif': Command execution failed","stack":"FileSystemError: Failed to read binary file '/workspace/template-helloworld/out/HelloWorld.gif': Command execution failed\n    at createErrorFromResponse (file:///Users/harshil/projects/claude-opus-video-generation/.wrangler/tmp/dev-r5cb9B/index.js:3489:14)\n    at FileClient.handleErrorResponse (file:///Users/harshil/projects/claude-opus-video-generation/.wrangler/tmp/dev-r5cb9B/index.js:3653:19)\n    at async FileClient.handleResponse (file:///Users/harshil/projects/claude-opus-video-generation/.wrangler/tmp/dev-r5cb9B/index.js:3623:23)\n    at async FileClient.readFile (file:///Users/harshil/projects/claude-opus-video-generation/.wrangler/tmp/dev-r5cb9B/index.js:3868:24)","name":"FileSystemError"}}


readFile failed, trying command-based fallback: Error: FileSystemError: Failed to read binary file '/workspace/template-helloworld/out/HelloWorld.gif': Command execution failed
    at async Array.<anonymous> (file:///Users/harshil/projects/claude-opus-video-generation/src/index.ts:178:20)
    at async jsonError (file:///Users/harshil/projects/claude-opus-video-generation/node_modules/wrangler/templates/middleware/middleware-miniflare3-json-error.ts:22:10)
    at async drainBody (file:///Users/harshil/projects/claude-opus-video-generation/node_modules/wrangler/templates/middleware/middleware-ensure-req-body-drained.ts:5:10) {
  remote: true
}
Attempting to read file using base64 command...
✘ [ERROR] Uncaught FileSystemError: Failed to read binary file '/workspace/template-helloworld/out/HelloWorld.gif': Command execution failed Error

      at createErrorFromResponse
  (file:///Users/harshil/projects/claude-opus-video-generation/node_modules/@cloudflare/sandbox/src/errors/adapter.ts:96:14)
      at handleErrorResponse
  (file:///Users/harshil/projects/claude-opus-video-generation/node_modules/@cloudflare/sandbox/src/clients/base-client.ts:179:19)


✘ [ERROR] {"level":"error","msg":"Unexpected error in execute","component":"sandbox-do","sandboxId":"747f7eee07b30243447c9845f40ca8064c4b3e112e06f821ea107ef045a2cda9","traceId":"tr_3b30b96af2674e30","httpStatus":500,"timestamp":"2025-11-26T12:28:27.731Z","error":{"message":"Failed to execute command 'base64 -w 0 \"/workspace/template-helloworld/out/HelloWorld.gif\"' in session 'sandbox-demo': Output too large: 23580272 bytes (max: 10485760)","stack":"CommandError: Failed to execute command 'base64 -w 0 \"/workspace/template-helloworld/out/HelloWorld.gif\"' in session 'sandbox-demo': Output too large: 23580272 bytes (max: 10485760)\n    at createErrorFromResponse (file:///Users/harshil/projects/claude-opus-video-generation/.wrangler/tmp/dev-r5cb9B/index.js:3496:14)\n    at CommandClient.handleErrorResponse (file:///Users/harshil/projects/claude-opus-video-generation/.wrangler/tmp/dev-r5cb9B/index.js:3653:19)\n    at async CommandClient.handleResponse (file:///Users/harshil/projects/claude-opus-video-generation/.wrangler/tmp/dev-r5cb9B/index.js:3623:23)\n    at async CommandClient.execute (file:///Users/harshil/projects/claude-opus-video-generation/.wrangler/tmp/dev-r5cb9B/index.js:3767:24)\n    at async Sandbox.execWithSession (file:///Users/harshil/projects/claude-opus-video-generation/.wrangler/tmp/dev-r5cb9B/index.js:5343:26)","name":"CommandError"}}


✘ [ERROR] Both read methods failed: Error: CommandError: Failed to execute command 'base64 -w 0 "/workspace/template-helloworld/out/HelloWorld.gif"' in session 'sandbox-demo': Output too large: 23580272 bytes (max: 10485760)

      at async Array.<anonymous>
  (file:///Users/harshil/projects/claude-opus-video-generation/src/index.ts:193:30)
      at async jsonError
  (file:///Users/harshil/projects/claude-opus-video-generation/node_modules/wrangler/templates/middleware/middleware-miniflare3-json-error.ts:22:10)
      at async drainBody
  (file:///Users/harshil/projects/claude-opus-video-generation/node_modules/wrangler/templates/middleware/middleware-ensure-req-body-drained.ts:5:10)
  {
    remote: true
  }


[wrangler:info] POST /vibe/transfer 500 Internal Server Error (1920ms)
✘ [ERROR] Uncaught CommandError: Failed to execute command 'base64 -w 0 "/workspace/template-helloworld/out/HelloWorld.gif"' in session 'sandbox-demo': Output too large: 23580272 bytes (max: 10485760) Error

      at createErrorFromResponse
  (file:///Users/harshil/projects/claude-opus-video-generation/node_modules/@cloudflare/sandbox/src/errors/adapter.ts:110:14)
      at handleErrorResponse
  (file:///Users/harshil/projects/claude-opus-video-generation/node_modules/@cloudflare/sandbox/src/clients/base-client.ts:179:19)
```
</details>

--- a/.changeset/remove-output-size-limit.md
+++ b/.changeset/remove-output-size-limit.md
@@ -1,0 +1,7 @@
+---
+'@cloudflare/sandbox': patch
+---
+
+Remove output size limit for command execution
+
+The 10MB output size limit that was intended to prevent OOM attacks has been removed. This limit was too restrictive for legitimate use cases like reading large media files. Developers are now trusted to manage their own resource usage and handle potential OOM situations.

--- a/docs/SESSION_EXECUTION.md
+++ b/docs/SESSION_EXECUTION.md
@@ -73,7 +73,6 @@ mkfifo "$sp" "$ep"
 ## Error Handling and Limits
 
 - Invalid `cwd` (foreground): we write a prefixed stderr line (binary prefix) indicating the failure and return exit code `1`.
-- Output size limit: large logs are rejected during parsing to protect memory (`MAX_OUTPUT_SIZE_BYTES`).
 - Timeouts: foreground commands can be configured to time out; an error is raised if the exit file does not appear in time.
 
 ## Why Two Patterns?
@@ -83,7 +82,7 @@ mkfifo "$sp" "$ep"
 
 ## Testing Notes
 
-- Foreground tests cover silent commands (`cd`, variable assignment), error scenarios, multiline output, and size limits.
+- Foreground tests cover silent commands (`cd`, variable assignment), error scenarios, and multiline output.
 - Background/streaming tests cover concurrent output, stderr separation, and completion events.
 - The previous hang class was caused by FIFO open/close races in foreground on silent commands; process substitution removes this class entirely.
 

--- a/packages/sandbox-container/src/config.ts
+++ b/packages/sandbox-container/src/config.ts
@@ -38,18 +38,6 @@ const COMMAND_TIMEOUT_MS = (() => {
 })();
 
 /**
- * Maximum output size in bytes to prevent OOM attacks.
- * This is a security measure, not a timeout.
- *
- * Default: 10MB
- * Environment variable: MAX_OUTPUT_SIZE_BYTES
- */
-const MAX_OUTPUT_SIZE_BYTES = parseInt(
-  process.env.MAX_OUTPUT_SIZE_BYTES || String(10 * 1024 * 1024),
-  10
-);
-
-/**
  * Delay between chunks when streaming output.
  * This debounces file system watch events for better performance.
  *
@@ -68,7 +56,6 @@ export const CONFIG = {
   INTERPRETER_SPAWN_TIMEOUT_MS,
   INTERPRETER_EXECUTION_TIMEOUT_MS,
   COMMAND_TIMEOUT_MS,
-  MAX_OUTPUT_SIZE_BYTES,
   STREAM_CHUNK_DELAY_MS,
   DEFAULT_CWD
 } as const;

--- a/packages/sandbox-container/src/session.ts
+++ b/packages/sandbox-container/src/session.ts
@@ -58,9 +58,6 @@ export interface SessionOptions {
   /** Command timeout in milliseconds (overrides CONFIG.COMMAND_TIMEOUT_MS) */
   commandTimeoutMs?: number;
 
-  /** Maximum output size in bytes (overrides CONFIG.MAX_OUTPUT_SIZE_BYTES) */
-  maxOutputSizeBytes?: number;
-
   /** Logger instance for structured logging (optional - uses no-op logger if not provided) */
   logger?: Logger;
 }
@@ -119,7 +116,6 @@ export class Session {
   private readonly id: string;
   private readonly options: SessionOptions;
   private readonly commandTimeoutMs: number | undefined;
-  private readonly maxOutputSizeBytes: number;
   private readonly logger: Logger;
   /** Map of running commands for tracking and killing */
   private runningCommands = new Map<string, CommandHandle>();
@@ -129,8 +125,6 @@ export class Session {
     this.options = options;
     this.commandTimeoutMs =
       options.commandTimeoutMs ?? CONFIG.COMMAND_TIMEOUT_MS;
-    this.maxOutputSizeBytes =
-      options.maxOutputSizeBytes ?? CONFIG.MAX_OUTPUT_SIZE_BYTES;
     // Use provided logger or create no-op logger (for backward compatibility/tests)
     this.logger = options.logger ?? createNoOpLogger();
   }
@@ -957,13 +951,6 @@ export class Session {
 
     if (!(await file.exists())) {
       return { stdout: '', stderr: '' };
-    }
-
-    // Safety check: prevent OOM attacks
-    if (file.size > this.maxOutputSizeBytes) {
-      throw new Error(
-        `Output too large: ${file.size} bytes (max: ${this.maxOutputSizeBytes})`
-      );
     }
 
     const content = await file.text();

--- a/packages/sandbox-container/tests/session.test.ts
+++ b/packages/sandbox-container/tests/session.test.ts
@@ -267,8 +267,8 @@ describe('Session', () => {
       ).toHaveLength(100);
     });
 
-    it('should handle output within size limit', async () => {
-      // Generate ~5KB of output (well below the 10MB default)
+    it('should handle large output without size limits', async () => {
+      // Generate ~5KB of output (no longer limited)
       const result = await session.exec(
         'yes "test line with some text" | head -n 500'
       );
@@ -280,28 +280,17 @@ describe('Session', () => {
       expect(lines.length).toBe(500);
     });
 
-    it('should reject output that exceeds max size', async () => {
-      // Create session with 1KB limit
-      const smallSession = new Session({
-        id: 'small-output-session',
-        cwd: testDir,
-        maxOutputSizeBytes: 1024 // 1KB
-      });
-      await smallSession.initialize();
+    it('should handle very large output without errors', async () => {
+      // Generate >1KB of output (~12KB) - should work without issues
+      const result = await session.exec(
+        'yes "test line with text here" | head -n 1000'
+      );
 
-      try {
-        // Try to generate >1KB of output (~12KB)
-        await smallSession.exec(
-          'yes "test line with text here" | head -n 1000'
-        );
-        expect.unreachable('Should have thrown an error for oversized output');
-      } catch (error) {
-        expect(error).toBeInstanceOf(Error);
-        expect((error as Error).message).toContain('Output too large');
-        expect((error as Error).message).toContain('1024');
-      } finally {
-        await smallSession.destroy();
-      }
+      expect(result.exitCode).toBe(0);
+      const lines = result.stdout
+        .split('\n')
+        .filter((l) => l === 'test line with text here');
+      expect(lines.length).toBe(1000);
     });
 
     it('should handle commands with special characters', async () => {

